### ISSUE TITLE
docker module exits with error while comparing lxc_options of container

### ISSUE THREAD

**Issue by ansibot** [ansible/ansible#30140]
##### ISSUE TYPE

bug report

##### COMPONENT NAME

docker module

##### ANSIBLE VERSION

1.9.0.1

##### OS / ENVIRONMENT

Ubuntu 14.04

##### SUMMARY

When ansible tries to compare lxc_conf options of a running container it exits with error:

```
Traceback (most recent call last):
  File "/root/.ansible/tmp/ansible-tmp-1430250632.51-188055754835863/docker", line 3106, in <module>
    main()
  File "/root/.ansible/tmp/ansible-tmp-1430250632.51-188055754835863/docker", line 1482, in main
    reloaded(manager, containers, count, name)
  File "/root/.ansible/tmp/ansible-tmp-1430250632.51-188055754835863/docker", line 1357, in reloaded
    for container in manager.get_differing_containers():
  File "/root/.ansible/tmp/ansible-tmp-1430250632.51-188055754835863/docker", line 981, in get_differing_containers
    expected_lxc = set(self.lxc_conf)
TypeError: unhashable type: 'dict'
```

`self.lxc_conf` and `container['HostConfig']['LxcConf']` look like an array of dict's, e.g.:

``` python
[{'Value': 'veth', 'Key': 'lxc.network.type'},
 {'Value': 'up', 'Key': 'lxc.network.flags'},
 {'Value': 'eth0', 'Key': 'lxc.network.name'}]
```

Comparing them as dict's works ok, it seems (modules/core/cloud/docker/docker.py:981):

``` python
                expected_lxc = dict([(lxc_option['Key'],lxc_option['Value']) for lxc_option in self.lxc_conf])
                actual_lxc = dict([(lxc_option['Key'],lxc_option['Value']) for lxc_option in container['HostConfig']['LxcConf'] or []])
```

Copied from original issue: ansible/ansible-modules-core#1220

**Comment by ansibot** [comment]
From @ansibot on 2015-04-28T20:08:28Z



@cove, @softzilla, @zfil, @ThomasSteinbach, ping. This issue is waiting on your response.
[click here for bot help](https://github.com/jctanner/ansibullbot/blob/ISSUE_TRIAGE_CLASS/ISSUE_HELP.md)

**Comment by biomassives** [comment]
This message was created automatically by mail delivery software.

A message that you sent could not be delivered to one or more of its
recipients. This is a temporary error. The following address(es) deferred:

  acmeideal@gmail.com
    Domain biomassiv.es has exceeded the max emails per hour (112/100 (112%)) allowed.  Message will be reattempted later

------- This is a copy of the message, including all the headers. ------
Received: from github-smtp2-ext3.iad.github.net ([192.30.252.194]:37166 helo=github-smtp2b-ext-cp1-prd.iad.github.net)
	by chi-server32.websitehostserver.net with esmtps (TLSv1.2:ECDHE-RSA-AES256-GCM-SHA384:256)
	(Exim 4.89)
	(envelope-from <noreply@github.com>)
	id 1droTl-001PpE-Dc
	for greg@biomassiv.es; Tue, 12 Sep 2017 11:56:05 -0500
Date: Tue, 12 Sep 2017 09:55:25 -0700
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=github.com;
	s=pf2014; t=1505235325;
	bh=/7yN7CT2fmoOEeVf6nSY+tCAnJNqvA//c2Xv4gOZwQ8=;
	h=From:Reply-To:To:Cc:In-Reply-To:References:Subject:List-ID:
	 List-Archive:List-Post:List-Unsubscribe:From;
	b=gJJbA9MY4T383NT+T5VJUaCNm4zkm6QYkgys5V8n9wuWpeIR90VhIMrWi06SScEae
	 s/YtAU7L/GsR+TVfdukZqFSUYNJyql08CE+Mxar9MRboiF71paS35UVeq5RtxcgPrY
	 248Z1yN2LYxuSGN8viGFk+80t+iLSXd+xFLvz2S8=

**Comment by ansibot** [comment]
@ansibot Greetings! Thanks for taking the time to open this issue. In order for the community to handle your issue effectively, we need a bit more information. 

Here are the items we could not find in your description:
- component name

Please set the description of this issue with this template:
https://raw.githubusercontent.com/ansible/ansible/devel/.github/ISSUE_TEMPLATE.md

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
cc @dusdanig @joshuaconner
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by gundalow** [comment]
The `cloud/docker/_docker.py` no longer exists in Ansible 2.5.0.

For information about the newer Docker modules to use, please see http://docs.ansible.com/ansible/devel/scenario_guides/guide_docker.html

If after reviewing the newer modules you believe this issue/PR would still be useful addition to the newer modules, please create a *new* issue/PR against the new module and reference this issue/PR in the comments.

Closing this. Any comments on here will be ignored.
